src/cdk-experimental/menu/menu-bar.spec.ts

```diff
@@ -0,0 +1,175 @@
+import {ComponentFixture, TestBed, async} from '@angular/core/testing';
+import {Component} from '@angular/core';
+import {CdkMenuBar} from './menu-bar';
+import {CdkMenuModule} from './menu-module';
+import {CdkMenuItem} from './menu-item';
+import {By} from '@angular/platform-browser';
+
+describe('MenuBar', () => {
+  describe('as radio group', () => {
+    let fixture: ComponentFixture<MenuBarRadioGroup>;
+    let menuBar: CdkMenuBar;
+    let menuItems: Array<CdkMenuItem>;
+
+    beforeEach(async(() => {
+      TestBed.configureTestingModule({
+        imports: [CdkMenuModule],
+        declarations: [MenuBarRadioGroup],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(MenuBarRadioGroup);
+      fixture.detectChanges();
+
+      menuBar = fixture.debugElement.query(By.directive(CdkMenuBar)).injector.get(CdkMenuBar);
+
+      menuItems = menuBar._allItems.toArray();
+    }));
+
+    it('should toggle menuitemradio items', () => {
+      expect(menuItems[0].checked).toBeTrue();
+      expect(menuItems[1].checked).toBeFalse();
+
+      menuItems[1].trigger();
+
+      expect(menuItems[0].checked).toBeFalse();
+      expect(menuItems[1].checked).toBeTrue();
+    });
+  });
+
+  describe('as checkbox group', () => {
+    let fixture: ComponentFixture<MenuBarCheckboxGroup>;
+    let menuBar: CdkMenuBar;
+    let menuItems: Array<CdkMenuItem>;
+
+    beforeEach(async(() => {
+      TestBed.configureTestingModule({
+        imports: [CdkMenuModule],
+        declarations: [MenuBarCheckboxGroup],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(MenuBarCheckboxGroup);
+
+      menuBar = fixture.debugElement.query(By.directive(CdkMenuBar)).injector.get(CdkMenuBar);
+      fixture.detectChanges();
+
+      menuItems = menuBar._allItems.toArray();
+    }));
+
+    it('should toggle menuitemcheckbox', () => {
+      expect(menuItems[0].checked).toBeTrue();
+      expect(menuItems[1].checked).toBeFalse();
+
+      menuItems[1].trigger();
```

Comment:
I'm not familiar with the public Angular testing conventions. Would this be the appropriate point to bring in test harnesses? (Would they apply for CDK directives, as opposed to actual components?)

Usually, it would be better to simulate real clicks using harness actions or mouse events, instead of directly calling methods on the directive class. But I don't know if that applies here, or if it's possible at this point in development.

Reply:
We actually don't use harnesses for the component's unit test; we create separate tests for the harnesses once we make them